src/Stryker.Core/Stryker.Core/TestRunners/VsTest/RunEventHandler.cs

```diff
@@ -203,6 +230,11 @@ public bool Wait(int timeOut)
                     Monitor.Wait(_lck, Math.Max(0, (int)(timeOut-watch.ElapsedMilliseconds)));
                 }
 
+                slept = watch.ElapsedMilliseconds - timeOut > 30 * 1000;
```

Comment:
Hmm why 30 mins? I mean, this is a configurable thing in an operated system... Shouldn't this info be somehow extracted from the OS?

Reply:
this is 30 seconds. it looks reasonable, any sleep of more than 30 seconds will be detected and invalidate this test run

Reply:
if Stryker waited 30 seconds more than the time out, it means either that the machine was really really busy (unlikely) or went to sleep (likely), so the timeout triggered as soon as the computer woke up (timeout are wall clock based).
Then we assume this may be a spurious timeout and it deserves a new try. I think any value below a few seconds may be risky, as a significant swap or context switch in a VM, or a GC, can lead to a few seconds of freeze.
On the other hand, if the user is next to his/her computer, there is a good chance he/she acts if/when it sees it going to sleep, meaning the computer maybe asleep for a few seconds before waking up.
I think the lowest sensible value is 10 seconds, and I think anything above a minute is too long.
hard to pick a good value.
Of course, having a definitive way to detect sleep mode would be better.

Reply:
> this is 30 seconds

Sorry, brain fart :D

> Of course, having a definitive way to detect sleep mode would be better.

I see, so that is a heuristic basically. Well, probably that's the best we can do. I mean looks like there is [something for that](https://stackoverflow.com/questions/16747146/c-sharp-sleep-detection-powermodechanged) but this doesn't seem to be cross-platform and yeah well accessing system APIs might lead to all sorts of access rights issues and so on.

Reply:
took a look at the doc. The events also need a message pump active; I think my hack is better as: 
1. generally available
2. identify what matters (a suspicious delay)

Reply:
Yep that's what I say, this is fine then.